src/passes/I64ToI32Lowering.cpp

```diff
@@ -0,0 +1,1197 @@
+      highBits,
+      builder->makeGetGlobal(highBitsGlobal, i32)
+    );
+    GetLocal* getLow = builder->makeGetLocal(lowBits, i32);
+    Block* result = builder->blockify(doCall, setHigh, getLow);
+    freeTemp(lowBits);
+    setOutParam(result, highBits);
+    replaceCurrent(result);
+  }
+  void visitCall(Call* curr) {
+    visitGenericCall<Call>(
+      curr,
+      [&](std::vector<Expression*>& args, WasmType ty) {
+        return builder->makeCall(curr->target, args, ty);
+      }
+    );
+  }
+
+  void visitCallImport(CallImport* curr) {
+    // imports cannot contain i64s
+    return;
+  }
+
+  void visitCallIndirect(CallIndirect* curr) {
+    visitGenericCall<CallIndirect>(
+      curr,
+      [&](std::vector<Expression*>& args, WasmType ty) {
+        return builder->makeCallIndirect(
+          curr->fullType,
+          curr->target,
+          args,
+          ty
+        );
+      }
+    );
+  }
+
+  void visitGetLocal(GetLocal* curr) {
+    if (curr->type != i64) return;
+    curr->index = indexMap[curr->index];
+    curr->type = i32;
+    Index highBits = getTemp();
+    SetLocal *setHighBits = builder->makeSetLocal(
+      highBits,
+      builder->makeGetLocal(
+        curr->index + 1,
+        i32
+      )
+    );
+    Block* result = builder->blockify(setHighBits, curr);
+    replaceCurrent(result);
+    setOutParam(result, highBits);
+  }
+
+  void lowerTee(SetLocal* curr) {
+    Index highBits = fetchOutParam(curr->value);
+    Index tmp = getTemp();
+    curr->index = indexMap[curr->index];
+    curr->type = i32;
+    SetLocal* setLow = builder->makeSetLocal(tmp, curr);
+    SetLocal* setHigh = builder->makeSetLocal(
+      curr->index + 1,
+      builder->makeGetLocal(highBits, i32)
+    );
+    GetLocal* getLow = builder->makeGetLocal(tmp, i32);
+    Block* result = builder->blockify(setLow, setHigh, getLow);
+    replaceCurrent(result);
+    setOutParam(result, highBits);
+    freeTemp(tmp);
+  }
+
+  void visitSetLocal(SetLocal* curr) {
+    if (!hasOutParam(curr->value)) return;
+    if (curr->isTee()) {
+      lowerTee(curr);
+      return;
+    }
+    Index highBits = fetchOutParam(curr->value);
+    curr->index = indexMap[curr->index];
+    SetLocal* setHigh = builder->makeSetLocal(
+      curr->index + 1,
+      builder->makeGetLocal(highBits, i32)
+    );
+    Block* result = builder->blockify(curr, setHigh);
+    replaceCurrent(result);
+    freeTemp(highBits);
+  }
+
+  void visitGetGlobal(GetGlobal* curr) {
+    assert(false && "GetGlobal not implemented");
+  }
+
+  void visitSetGlobal(SetGlobal* curr) {
+    assert(false && "SetGlobal not implemented");
+  }
+
+  void visitLoad(Load* curr) {
+    if (curr->type != i64) return;
+    assert(!curr->isAtomic && "atomic load not implemented");
+    Index highBits = getTemp();
+    Index ptrTemp = getTemp();
+    SetLocal* setPtr = builder->makeSetLocal(ptrTemp, curr->ptr);
+    SetLocal* loadHigh;
+    if (curr->bytes > 4) {
```

Comment:
might as well make it `== 8`

Reply:
I think so, wasm only supports 1, 2, 4, and 8 byte sized loads unless something big changed recently that I am unaware of.

Reply:
Is this also the case for stores?

Reply:
@kripken @jgravelle-google Bump. I'd like to get this quick fix in so this can be merged. Is it both loads and stores that only support 1, 2, 4, or 8 bytes?

Reply:
Looks pretty symmetric: https://github.com/WebAssembly/design/blob/master/Semantics.md#linear-memory-accesses